phat/contracts/lego/js/src/index.ts

```diff
@@ -1,190 +1,279 @@
-import './polyfills.js'
-import '@phala/pink-env'
-import * as scaleCore from '@scale-codec/core'
+import "@phala/pink-env";
+import { TypeRegistry } from "@phala/pink-env";
 
-const pink = globalThis.pink
-
-type HexString = `0x${string}`
-interface BaseAction {
-  name?: string
-  input?: any
+declare global {
+  var debugWorkflow: boolean;
 }
 
-interface ActionLog extends BaseAction {
-  cmd: 'log'
+function doEval(script: string, input: any, context: any): any {
+  function numToUint8Array32(num: number) {
+    let arr = new Uint8Array(32);
+    for (let i = 0; i < 32; i++) {
+      arr[i] = num % 256;
+      num = Math.floor(num / 256);
+    }
+    return arr;
+  }
```

Comment:
Good `asm.js` style. Do you mean `| 0` rather than `| 1`?

Given the num is usually already an integer, I think there should not much advantage on performance to this style but would pay for the extra `|` operations.